packages/babel-traverse/src/path/replacement.js

```diff
@@ -212,10 +212,14 @@ export function replaceExpressionWithStatements(nodes: Array<Object>) {
 
       let loop = path.findParent((path) => path.isLoop());
       if (loop) {
-        let callee = this.get("callee");
-
-        let uid = callee.scope.generateDeclaredUidIdentifier("ret");
-        callee.get("body").pushContainer("body", t.returnStatement(uid));
+        let uid = loop.getData("expressionReplacementReturnUid");
+
+        if (!uid) {
+          const callee = this.get("callee");
+          uid = callee.scope.generateDeclaredUidIdentifier("ret");
+          callee.get("body").pushContainer("body", t.returnStatement(uid));
+          loop.setData("expressionReplacementReturnUid", uid);
+        }
```

Comment:
We should probably do

```
} else {
  uid = t.identifier(uid.name);
}
```
or else we'll be inserting the same identifier node in multiple places in the AST.